api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -116,9 +116,39 @@ public interface ExpireSnapshots extends PendingUpdate<List<Snapshot>> {
    *
    * @param clean setting this to false will skip deleting expired manifests and files
    * @return this for method chaining
+   * @deprecated since 1.10.0, will be removed in 2.0.0; use {@link #cleanMode(CleanupMode)}
+   *     instead.
    */
+  @Deprecated
   ExpireSnapshots cleanExpiredFiles(boolean clean);
 
+  /**
+   * Configures the cleanup mode for expired files.
+   *
+   * <p>This method provides fine-grained control over which files are cleaned up during snapshot
+   * expiration. The cleanup modes are:
+   *
+   * <ul>
+   *   <li>{@link CleanupMode#ALL} - Clean up both metadata and data files (default)
+   *   <li>{@link CleanupMode#METADATA_ONLY} - Clean up only metadata files (manifests, manifest
```

Comment:
metadata only will also cleanup statistics files as well, so I think we should call that out.